This patch is based on the upstream commit described below, adapted for use
in the Debian package by Peter Michael Green.

commit aaadde7fc006cca048691aa0bb5dfd2d1d7eed4a
Author: Guillaume Gomez <guillaume.gomez@huawei.com>
Date:   Tue Jan 18 17:34:33 2022 +0100

    Update to sysinfo 0.23

Index: process-viewer/src/display_procs.rs
===================================================================
--- process-viewer.orig/src/display_procs.rs
+++ process-viewer/src/display_procs.rs
@@ -7,7 +7,7 @@ use gtk::prelude::{
 };
 use gtk::{self, glib};
 
-use sysinfo::{AsU32, Pid, Process, ProcessExt};
+use sysinfo::{Pid, PidExt, Process, ProcessExt};
 
 use crate::notebook::NoteBook;
 use crate::utils::{create_button_with_image, format_number};
@@ -105,7 +105,7 @@ impl Procs {
                 let selection = tree_view.selection();
                 let (pid, ret) = if let Some((model, iter)) = selection.selected() {
                     if let Ok(x) = model.value(&iter, 0).get::<u32>() {
-                        (Some(x as Pid), true)
+                        (Some(Pid::from_u32(x)), true)
                     } else {
                         (None, false)
                     }
Index: process-viewer/src/main.rs
===================================================================
--- process-viewer.orig/src/main.rs
+++ process-viewer/src/main.rs
@@ -59,7 +59,7 @@ fn update_window(list: &gtk::ListStore,
         let mut valid = true;
         while valid {
             let pid = match list.value(&iter, 0).get::<u32>() {
-                Ok(pid) => pid as Pid,
+                Ok(pid) => Pid::from_u32(pid),
                 _ => {
                     valid = list.iter_next(&iter);
                     continue;
@@ -466,7 +466,7 @@ fn build_ui(application: &gtk::Applicati
                 let pid = model.value(&iter, 0)
                                .get::<u32>()
                                .expect("Model::get failed");
-                create_new_proc_diag(&process_dialogs, pid as Pid, &*sys.lock().expect("failed to lock to create new proc dialog (from tree)"));
+                create_new_proc_diag(&process_dialogs, Pid::from_u32(pid), &*sys.lock().expect("failed to lock to create new proc dialog (from tree)"));
             }
         ));
 
Index: process-viewer/Cargo.toml
===================================================================
--- process-viewer.orig/Cargo.toml
+++ process-viewer/Cargo.toml
@@ -33,7 +33,7 @@ version = "1.0"
 version = "1.0"
 
 [dependencies.sysinfo]
-version = "0.22.3"
+version = "0.23.0"
 
 [dependencies.toml]
 version = "0.5"
